src/m365/teams/commands/chat/chat-member-add.ts

```diff
@@ -0,0 +1,143 @@
+        includeAllHistory: !!args.options.includeAllHistory
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --chatId <chatId>'
+      },
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userName [userName]'
+      },
+      {
+        option: '--role [role]',
+        autocomplete: TeamsChatMemberAddCommand.roles
+      },
+      {
+        option: '--visibleHistoryStartDateTime [visibleHistoryStartDateTime]'
+      },
+      {
+        option: '--includeAllHistory'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (!validation.isValidTeamsChatId(args.options.chatId)) {
+          return `${args.options.chatId} is not a valid chatId.`;
+        }
+
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid userId.`;
+        }
+
+        if (args.options.userName && !validation.isValidUserPrincipalName(args.options.userName)) {
+          return `${args.options.userName} is not a valid userName.`;
+        }
+
+        if (args.options.role && TeamsChatMemberAddCommand.roles.indexOf(args.options.role) < 0) {
+          return `${args.options.role} is not a valid role. Allowed values are ${TeamsChatMemberAddCommand.roles.join(', ')}`;
+        }
+
+        if (args.options.visibleHistoryStartDateTime && !validation.isValidISODateTime(args.options.visibleHistoryStartDateTime)) {
+          return `'${args.options.visibleHistoryStartDateTime}' is not a valid visibleHistoryStartDateTime.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['userId', 'userName'] },
+      {
+        options: ['visibleHistoryStartDateTime', 'includeAllHistory'], runsWhen(args) {
+          return args.options.visibleHistoryStartDateTime || args.options.includeAllHistory;
+        }
+      });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      if (this.verbose) {
+        logger.logToStderr(`Adding member ${args.options.userId || args.options.userName} to chat with id ${args.options.chatId}...`);
+      }
+
+      const chatMemberRemoveOptions: CliRequestOptions = {
+        url: `${this.resource}/v1.0/chats/${args.options.chatId}/members`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json',
+        data: {
+          '@odata.type': '#microsoft.graph.aadUserConversationMember',
+          'user@odata.bind': `https://graph.microsoft.com/v1.0/users/${args.options.userId || args.options.userName}`,
+          visibleHistoryStartDateTime: args.options.visibleHistoryStartDateTime || args.options.includeAllHistory ? '0001-01-01T00:00:00Z' : undefined,
```

Comment:
Currently, when specifying a specific date, we are exposing all the chat history.